object/id/address.go

```diff
@@ -0,0 +1,148 @@
+package oid
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/nspcc-dev/neofs-api-go/v2/refs"
+	cid "github.com/nspcc-dev/neofs-sdk-go/container/id"
+)
+
+// Address represents global object identifier in NeoFS network. Each object
+// belongs to exactly one container and is uniquely addressed within the container.
+//
+// Address is mutually compatible with github.com/nspcc-dev/neofs-api-go/v2/refs.Address
+// message. See ReadFromV2 / WriteToV2 methods.
+//
+// Instances can be created using built-in var declaration.
+//
+// Note that direct typecast is not safe and may result in loss of compatibility:
+// 	_ = Address(refs.Address{}) // not recommended
+type Address struct {
+	cnr cid.ID
+
+	obj ID
+}
+
+// ReadFromV2 reads Address from the refs.Address message. Returns an error if
+// the message is malformed according to the NeoFS API V2 protocol.
+//
+// See also WriteToV2.
+func (x *Address) ReadFromV2(m refs.Address) error {
+	cnr := m.GetContainerID()
+	if cnr == nil {
+		return errors.New("missing container ID")
+	}
+
+	obj := m.GetObjectID()
+	if obj == nil {
+		return errors.New("missing object ID")
+	}
+
+	err := x.cnr.ReadFromV2(*cnr)
+	if err != nil {
+		return fmt.Errorf("invalid container ID: %w", err)
+	}
+
+	err = x.obj.ReadFromV2(*obj)
+	if err != nil {
+		return fmt.Errorf("invalid object ID: %w", err)
+	}
+
+	return nil
+}
+
+// WriteToV2 writes Address to the refs.Address message.
+// The message must not be nil.
+//
+// See also ReadFromV2.
+func (x Address) WriteToV2(m *refs.Address) {
+	var obj refs.ObjectID
+	x.obj.WriteToV2(&obj)
+
+	var cnr refs.ContainerID
+	x.cnr.WriteToV2(&cnr)
+
+	m.SetObjectID(&obj)
+	m.SetContainerID(&cnr)
+}
+
+// Container sets unique identifier of the NeoFS object container.
+//
+// Zero Address has zero container ID, which is incorrect according to NeoFS
+// API protocol.
+//
+// See also SetContainer.
+func (x Address) Container() cid.ID {
```

Comment:
why have you decided not to use "ID" in all getters and setters?

Reply:
Since we obviously deal with identifiers while working with `Address`.